### PR TITLE
Added feature: Caching using query parameter

### PR DESCRIPTION
**Pull request by rgabs**

### CODE REVIEW

File: app/scripts/urlHandler.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    return {
+        setItem,
+        getItems,
+        hasItems,
+        removeItem,
+        resetItems
+    }
+
+})();
```
Comment (himanshusingh2407) [inline app/scripts/urlHandler.js:48]: Please add an empty line at the end of this file as followed in `audioScript.js` and `main.js`.

```suggestion
})();

```

File: app/scripts/urlHandler.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            const decodedURI = decodeURI(window.location.hash).split('#')[1];
+            return JSON.parse(decodedURI);
+        }
+        catch(e) {
+            return {};
+        }
+    }
+
+    function resetItems() {
```
Comment (himanshusingh2407) [inline app/scripts/urlHandler.js]: `resetItems` is exposed but not used anywhere.

File: app/index.html [hunk 3]
```diff
@@ -85,6 +85,11 @@
         </div>
         <div class="text">Give Feedback</div>
       </div>
+
+      <a class="menu-option" href="/">
```
Comment (himanshusingh2407) [inline app/index.html]: `href="/"` will always reload the page. We can do `onclick="pauseAll()"` instead to pause all.

**Review by himanshusingh2407 — CHANGES_REQUESTED** [review]

**Comment by himanshusingh2407** [comment]
Thanks @rgabs for taking the time to build this feature. 
👏🏻

File: app/scripts/urlHandler.js (hunk 1 above)
Comment (rgabs) [reply app/scripts/urlHandler.js:48]: why is that required though?

**Pull request merged**